modules/platforms/dotnet/Apache.Ignite.Core/Impl/Transactions/TransactionRollbackOnlyProxy.cs

```diff
@@ -0,0 +1,252 @@
+        public TransactionConcurrency Concurrency { get; private set; }
+
+        /** <inheritdoc /> */
+        public TransactionState State
+        {
+            get
+            {
+                lock (this)
+                {
+                    ThrowIfClosed();
+
+                    return _txs.TxState(this);   
+                }
+            }
+        }
+
+        /** <inheritdoc /> */
+        public TimeSpan Timeout { get; private set; }
+
+        /** <inheritdoc /> */
+        public string Label { get; private set; }
+
+        /** <inheritdoc /> */
+        public bool IsRollbackOnly
+        {
+            get { return true; }
+        }
+
+        /** <inheritDoc /> */
+        public bool SetRollbackonly()
+        {
+            throw GetInvalidOperationException();
+        }
+
+        /** <inheritDoc /> */
+        public void Commit()
+        {
+            throw GetInvalidOperationException();
+        }
+
+        /** <inheritDoc /> */
+        public Task CommitAsync()
+        {
+            throw GetInvalidOperationException();
+        }
+
+        /** <inheritDoc /> */
+        public void Rollback()
+        {
+            lock (this)
+            {
+                ThrowIfClosed();
+
+                try
+                {
+                    _txs.TxRollback(this);
+                }
+                finally
+                {
+                    _isClosed = true;
+                }
+            }
+        }
+
+        /** <inheritDoc /> */
+        public Task RollbackAsync()
+        {
+            lock (this)
+            {
+                ThrowIfClosed();
+
+                return _txs.TxRollbackAsync(this)
+                    .ContWith(t =>
+                    {
+                        try
+                        {
+                            _txs.TxClose(this);
+                        }
+                        finally
+                        {
+                            _isClosed = true;
+                        }
+                    });
+            }
+        }
+
+        /** <inheritDoc /> */
+        public void AddMeta<TV>(string name, TV val)
+        {
+            throw GetInvalidOperationException();
+        }
+
+        /** <inheritdoc /> */
+        public TV Meta<TV>(string name)
+        {
+            throw GetInvalidOperationException();
+        }
+
+        /** <inheritDoc /> */
+        public TV RemoveMeta<TV>(string name)
+        {
+            throw GetInvalidOperationException();
+        }
+
+        /** <inheritDoc /> */
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes",
+            Justification = "Dispose should not throw.")]
+        public void Dispose()
+        {
+            if (!_isClosed)
+            {
+                try
+                {
+                    _txs.TxRemove(this);
+                }
+                catch
+                {
+                    // No-op. Dispose should not throw.
```

Comment:
Same here, let's make sure we handle only known exceptions

Reply:
Fixed